src/SoftRobots.Inverse/binding/Binding_QPInverseProblemSolver.cpp

```diff
@@ -0,0 +1,143 @@
+#include <SofaPython3/PythonEnvironment.h>
+#include <SofaPython3/Sofa/Core/Binding_Base.h>
+
+#include <SoftRobots.Inverse/binding/Binding_QPInverseProblemSolver.h>
+#include <SoftRobots.Inverse/binding/Binding_QPInverseProblemSolver_doc.h>
+
+SOFAPYTHON3_BIND_ATTRIBUTE_ERROR()
+
+/// Makes an alias for the pybind11 namespace to increase readability.
+namespace py { using namespace pybind11; }
+
+namespace softrobotsinverse::python3
+{
+using sofa::core::objectmodel::Event;
+using sofa::core::objectmodel::BaseObject;
+using softrobotsinverse::solver::QPInverseProblemSolver;
+using EigenDenseMatrix = Eigen::Matrix<SReal, Eigen::Dynamic, Eigen::Dynamic>;
+using EigenMatrixMap = Eigen::Map<EigenDenseMatrix>;
+
+std::string QPInverseProblemSolver_Trampoline::getClassName() const
+{
+    sofapython3::PythonEnvironment::gil acquire {"getClassName"};
+    // Get the actual class name from python.
+    return py::str(py::cast(this).get_type().attr("__name__"));
+}
+
+bool QPInverseProblemSolver_Trampoline::solveSystem()
+{
+    PYBIND11_OVERLOAD_PURE(bool,
+                          QPInverseProblemSolver_Trampoline,
+                          solveSystem,
+                          );
+}
+
+bool QPInverseProblemSolver_Trampoline::solveSystem(const sofa::core::ConstraintParams* cParams,
+                                                    sofa::core::MultiVecId res1,
+                                                    sofa::core::MultiVecId res2)
+{
+    SOFA_UNUSED(cParams);
+    SOFA_UNUSED(res1);
+    SOFA_UNUSED(res2);
+
+    const bool result = solveSystem();
+    storeResults();
+
+    return result;
+}
+
+void QPInverseProblemSolver_Trampoline::storeResults()
+{
+    SReal *lambda = m_currentCP->getF();
+    SReal **w = m_currentCP->getW();
+    SReal *dfree = m_currentCP->getDfree();
+
+    solver::module::QPInverseProblem::QPConstraintLists* qpCLists = m_currentCP->getQPConstraintLists();
+
+    const std::size_t nbActuatorRows = qpCLists->actuatorRowIds.size();
+    const std::size_t nbEffectorRows = qpCLists->effectorRowIds.size();
+    const std::size_t nbSensorRows   = qpCLists->sensorRowIds.size();
+    const std::size_t nbContactRows  = qpCLists->contactRowIds.size();
+    const std::size_t nbEqualityRows = qpCLists->equalityRowIds.size();
+    const std::size_t nbRows = nbEffectorRows + nbActuatorRows + nbContactRows + nbSensorRows + nbEqualityRows;
+
+    solver::module::QPInverseProblemImpl::QPSystem* qpSystem = m_currentCP->getQPSystem();
+    qpSystem->delta.resize(nbRows);
+    for(std::size_t i=0; i<nbRows; i++)
+    {
+        qpSystem->delta[i] = dfree[i];
+        for(std::size_t j=0; j<nbRows; j++)
+            qpSystem->delta[i] += lambda[j]*w[i][j];
+    }
+
+    m_currentCP->sendResults();
+}
+
+void moduleAddQPInverseProblemSolver(py::module &m)
+{
+    py::class_<softrobotsinverse::solver::QPInverseProblemSolver,
+               QPInverseProblemSolver_Trampoline,
+               BaseObject,
```

Comment:
```suggestion
               ConstraintSolverImpl,
```
Try with this suggestion and remove the already-existing bindings

Reply:
Yes that was the problem, thanks